### PR TITLE
Enhancement of Logging Mechanisms for KS Cloud Usage

### LINKED ISSUE #1543: KSCloudAPIConnector possibly not working
# Description

I have just been testing for any vulnerabilities, and it turns out this command runs with no errors.
I am not sure if this is just not implemented yet or a bug from the KSClient

# Environment
OS: ` ` 
Version: ` ` 
 
# Steps To Reproduce
1. Run the following 
```kubescape list frameworks --account "22019933-feac-4012-a8eb-e81461ba3655" --access-key "wrongpassword"```

# Expected behavior
Should throw an error, as the access-key is wrong

# Actual Behavior
Returns result, as if the arguments are not used
image

# Additional context
This happens when the env var `KS_CLOUD_API_URL` is not set, or when a timeout/auth failure happens

### PR DESCRIPTION
**Pull request by mohaidoss**
Closes #1543 

## type:
Enhancement

___
## description:
This PR introduces enhanced logging mechanisms for when KS Cloud is used. The changes include:
- Logging a warning when the Kubescape Cloud API URL is not set and policies are loaded from cache.
- Logging a warning when there's a failure in listing frameworks.

___
## main_files_walkthrough:
<details> <summary>files:</summary>

- `core/core/initutils.go`: Added two logging warnings:
- When the Kubescape Cloud API URL is not set and policies are loaded from cache.
- When there's a failure in listing frameworks.
</details>

### CODE REVIEW

File: core/core/initutils.go [hunk 1]
```diff
@@ -202,7 +202,10 @@ func getPolicyGetter(ctx context.Context, loadPoliciesFromFile []string, account
 	if accountID != "" && getter.GetKSCloudAPIConnector().GetCloudAPIURL() != "" && frameworkScope {
 		g := getter.GetKSCloudAPIConnector() // download policy from Kubescape Cloud backend
 		return g
+	} else if accountID != "" && getter.GetKSCloudAPIConnector().GetCloudAPIURL() == "" && frameworkScope {
+		logger.L().Ctx(ctx).Warning("Kubescape Cloud API URL is not set, loading policies from cache")
```
Comment (matthyx) [inline core/core/initutils.go:206]: I am not sure this is warning... wdyt @Daniel-GrunbergerCA  ?

Comment (amirmalka) [reply core/core/initutils.go:206]: Not sure it should be a warning either; maybe a debug print?

Comment (mohaidoss) [reply core/core/initutils.go:206]: I mean, if a user is providing his credentials. That essentially means that he is trying to connect to the Cloud API. 
The current behavior doesn't tell the user if he is actually using the API or not

Comment (Daniel-GrunbergerCA) [reply core/core/initutils.go:206]: I think it makes sense to put this as a `warning`

**Pull request merged**